src/Core/src/Primitives/SafeAreaEdges.cs

```diff
@@ -0,0 +1,171 @@
+		/// <param name="vertical">The value to apply to top and bottom edges.</param>
+		public SafeAreaEdges(SafeAreaRegions horizontal, SafeAreaRegions vertical)
+		{
+			Left = horizontal;
+			Right = horizontal;
+			Top = vertical;
+			Bottom = vertical;
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SafeAreaEdges"/> struct with individual values for each edge.
+		/// </summary>
+		/// <param name="left">The value for the left edge.</param>
+		/// <param name="top">The value for the top edge.</param>
+		/// <param name="right">The value for the right edge.</param>
+		/// <param name="bottom">The value for the bottom edge.</param>
+		public SafeAreaEdges(SafeAreaRegions left, SafeAreaRegions top, SafeAreaRegions right, SafeAreaRegions bottom)
+		{
+			Left = left;
+			Top = top;
+			Right = right;
+			Bottom = bottom;
+		}
+
+		/// <summary>
+		/// Gets the safe area behavior for the specified edge.
+		/// </summary>
+		/// <param name="edge">The edge index (0=Left, 1=Top, 2=Right, 3=Bottom).</param>
+		/// <returns>The <see cref="SafeAreaRegions"/> for the specified edge.</returns>
+		public SafeAreaRegions GetEdge(int edge)
+		{
+			return edge switch
+			{
+				0 => Left,
+				1 => Top,
+				2 => Right,
+				3 => Bottom,
+				_ => SafeAreaRegions.None
+			};
+		}
+
+		/// <summary>
+		/// Sets the safe area behavior for the specified edge.
+		/// </summary>
+		/// <param name="edge">The edge index (0=Left, 1=Top, 2=Right, 3=Bottom).</param>
+		/// <param name="value">The value to set.</param>
+		public void SetEdge(int edge, SafeAreaRegions value)
+		{
+			switch (edge)
+			{
+				case 0:
+					Left = value;
+					break;
+				case 1:
+					Top = value;
+					break;
+				case 2:
+					Right = value;
+					break;
+				case 3:
+					Bottom = value;
+					break;
+			}
+		}
+
+		/// <summary>
+		/// A <see cref="SafeAreaEdges"/> with all edges set to <see cref="SafeAreaRegions.None"/>.
+		/// </summary>
+		public static SafeAreaEdges Default => new(SafeAreaRegions.None);
+
+		/// <summary>
+		/// A <see cref="SafeAreaEdges"/> with all edges set to <see cref="SafeAreaRegions.None"/>.
+		/// </summary>
+		public static SafeAreaEdges None => new(SafeAreaRegions.None);
+
+		/// <summary>
+		/// A <see cref="SafeAreaEdges"/> with all edges set to <see cref="SafeAreaRegions.All"/>.
+		/// </summary>
+		public static SafeAreaEdges All => new(SafeAreaRegions.All);
+
+		public static bool operator ==(SafeAreaEdges left, SafeAreaEdges right)
+		{
+			return left.Equals(right);
+		}
+
+		public static bool operator !=(SafeAreaEdges left, SafeAreaEdges right)
+		{
+			return !left.Equals(right);
+		}
+
+		public bool Equals(SafeAreaEdges other)
+		{
+			return Left == other.Left && Top == other.Top && Right == other.Right && Bottom == other.Bottom;
+		}
+
+		public override bool Equals(object? obj)
+		{
+			return obj is SafeAreaEdges other && Equals(other);
+		}
+
+		public override int GetHashCode()
+		{
+			unchecked
```

Comment:
#if NET (net 5 and greater)
return HashCode.Combine(Left,Top,Right,Bottom);
#else